ID: 58119
Title: flaky test TestSyncLoadOnObjectWhichCanNotFoundInStorage
Description:
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tidb/pipelines/ghpr_unit_test/runs/24042/nodes/63/steps/69/log/?start=0
```
--- FAIL: TestSyncLoadOnObjectWhichCanNotFoundInStorage (29.45s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x86785d8]

goroutine 12 [running]:
testing.tRunner.func1.2({0x8d2cea0, 0xd0e2620})
	GOROOT/src/testing/testing.go:1632 +0x3fc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x6b6
panic({0x8d2cea0?, 0xd0e2620?})
	GOROOT/src/runtime/panic.go:785 +0x132
pkg/statistics/handle/syncload/syncload_test_test.TestSyncLoadOnObjectWhichCanNotFoundInStorage(0xc001141380)
	pkg/statistics/handle/syncload/stats_syncload_test.go:469 +0x12d8
testing.tRunner(0xc001141380, 0x96de128)
	GOROOT/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1743 +0x826
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)
master
<!-- Paste the output of SELECT tidb_version() -->

